File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int ans = nums[0];
        int cnt = 1;
        for(int i = 1; i < n; i++) {
            if(nums[i] == ans) {
                cnt++;
            }
            else cnt--;
            if(cnt == 0) {
                ans = nums[i + 1];
                cnt = 0;
            }
        }
        return ans;
    }
};

File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    ans, cnt := nums[0], 1
    for i := 1; i < len(nums); i++ {
        if nums[i] == ans {
            cnt++
        } else {
            cnt--
        }
        if cnt == 0 {
            ans = nums[i + 1]
            cnt = 0
        }
    }
    return ans
}